packages/abstract-core/src/objects/price_source.rs

```diff
@@ -247,3 +251,184 @@ fn query_cw20_supply(querier: &QuerierWrapper, contract_addr: &Addr) -> Abstract
         }))?;
     Ok(response.total_supply)
 }
+
+#[cfg(test)]
+mod tests {
+    use abstract_testing::prelude::*;
+    use cosmwasm_std::testing::mock_dependencies;
+    use speculoos::prelude::*;
+
+    use super::*;
+
+    // TODO: abstract_testing has a circular dependency with this package, and so the mockAns host is unable to be used.
+    mod check {
+        use crate::ans_host;
+        use cosmwasm_std::testing::mock_dependencies;
+
+        use crate::objects::pool_id::PoolAddressBase;
+
+        use super::*;
+
+        #[test]
+        fn liquidity_token() -> AbstractResult<()> {
+            let mut deps = mock_dependencies();
+            // we cannot use the MockAnsHost because it has a circular dependency with this package
```

Comment:
circular dependecies are accepted under two conditions: 
1. The dependency is a dev-dep
2. The source of the dependency is a `path`